api/oci/extensions/repositories/ocireg/repository.go

```diff
@@ -112,7 +115,7 @@ func (r *RepositoryImpl) getCreds(comp string) (credentials.Credentials, error)
 	return identity.GetCredentials(r.GetContext(), r.info.Locator, comp)
 }
 
-func (r *RepositoryImpl) getResolver(comp string) (resolve.Resolver, error) {
+func (r *RepositoryImpl) getResolver(ref string, comp string) (registry.Repository, error) {
 	creds, err := r.getCreds(comp)
 	if err != nil {
 		if !errors.IsErrUnknownKind(err, credentials.KIND_CONSUMER) {
```

Comment:
see comment above, instead, provide a second method used to prepare a pusher for a version push based on the .ociRepo attribute already resolved.